modules/exploits/linux/http/grandstream_gxv3175_settimezone_unauth_cmd_exec.rb

```diff
@@ -0,0 +1,108 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+
+  HttpFingerprint = { pattern: [ /Multimedia Phone/ ] }.freeze
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => "Grandstream GXV3175 'settimezone' Unauthenticated Command Execution",
+        'Description' => %q{
+          This module exploits a command injection vulnerability in Grandstream GXV3175
+          IP multimedia phones. The 'settimezone' action does not validate input in the
+          'timezone' parameter allowing injection of arbitrary commands.
+
+          A buffer overflow in the 'phonecookie' cookie parsing allows authentication
+          to be bypassed by providing an alphanumeric cookie 93 characters in length.
+
+          This module was tested successfully on Grandstream GXV3175v2
+          hardware revision V2.6A with firmware version 1.0.1.19.
+        },
+        'Author' => [
+          'alhazred', # Command injection vulnerability discovery and exploit
+          'Brendan Scarvell', # Auth bypass discovery
+          'bcoles' # Metasploit
+        ],
+        'License' => MSF_LICENSE,
+        'Platform' => 'linux',
+        'References' => [
+          [ 'CVE', '2019-10655' ],
+          [ 'URL', 'https://www.trustwave.com/en-us/resources/security-resources/security-advisories/?fid=23920' ],
+          [ 'URL', 'https://github.com/dirtyfilthy/gxv3175-remote-code-exec/blob/master/modules/exploits/linux/http/grandstream_gxv3175_cmd_exec.rb' ]
+        ],
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        },
+        'DisclosureDate' => '2014-09-17',
```

Comment:
This was from the original PoC right, but wasn't that only disclosed on September 1st, 2016?

Reply:
changed